doc/decisions/frontend_keyhandling.md

```diff
@@ -0,0 +1,65 @@
+# Frontend Keybindings
+
+## Problem
+
+In the development of PermaplanT, we have identified the need for a robust keybinding system to enhance the user experience. Keybindings are utilized in various parts of the application, and we need to establish a cohesive approach for their implementation.
+
+## Constraints
+
+Our keybinding solution should adhere to the following constraints:
+
+- **Ease of Integration**: The keybinding system should be straightforward to integrate into different components of the application.
+
+- **User Configuration**: We should plan for future functionality that allows users to configure keybindings according to their preferences.
+
+## Decision / Current Solution
+
+We have decided to configure keybindings using a JSON structure that includes scopes and associated keybindings.
+
+Example:
+
+```
+{
+  "global": {
+    "Ctrl+M": "navigateToMap"
+  }
+  "planting": {
+    "Escape": "exitPlantingMode"
+  }
+}
+```
+
+Components that require keybindings can define action handlers for the configured actions. Here's an example of how this can be implemented:
+
+Example:
+
+```
+  const keyHandlerActions : Record<string, () => void> = {
+    exitPlantingMode: () => {
+        exitPlantingMode();
+    }
+  }
+```
+
+We can then add key listeners to specific components, and if a key mapping is configured for the pressed key within the associated scope, the corresponding action can be executed.
+
+## Challenges / To be discussed
+
+### Where do we add key listeners?
+
+While it's easy to add key listeners to focused components like form inputs, it becomes more challenging when dealing with a component like the Konva map. This is because we cannot directly bind keys to the map layer.
+
+Furthermore there are instances where we want to handle keys for map feature even when another component is focused. For example, if a user selects a plant in the right toolbar, pressing the "Escape" key should probably also exit planting mode.
```

Comment:
What we want from UI perspective should be described in @tschawax document (#918)